.github/workflows/main.yml

```diff
@@ -0,0 +1,31 @@
+name: CI
+on:
+  push:
+    branches:
+      - 'main'
+      - 'versions/*'
+    tags:
+      - 'v*'
+  pull_request:
+    branches:
+      - '**'
+
+jobs:
+  npm-test:
+    uses: ./.github/workflows/npm-test.yml
+
+  docker:
+    needs: npm-test
+    if: startsWith(github.ref, 'refs/tags/v') || (github.ref == 'refs/heads/main') || startsWith(github.ref, 'refs/heads/versions/')
+    uses: ./.github/workflows/docker.yml
+    secrets:
+      DOCKERHUB_USERNAME: ${{ secrets.DOCKERHUB_USERNAME }}
+      DOCKERHUB_TOKEN: ${{ secrets.DOCKERHUB_TOKEN }}
+
+  typedocs:
+    if: startsWith(github.ref, 'refs/tags/v') && !contains(github.ref, '-')
+    uses: ./.github/workflows/typedocs.yml
+
+  mkdocs:
+    if: startsWith(github.ref, 'refs/tags/v') && !contains(github.ref, '-')
+    uses: ./.github/workflows/mkdocs.yml
```

Comment:
So I guess this is the regular flow, and the mkdocs being built on pushes to main are in the `workflow_call:` section of the `.github/workflows/mkdocs.yml` file?

Reply:
The mkdocs workflow has two triggers. One is `workflow_call` which is used in the CI, `main.yml`, workflow. The other is a simple `push` which only gets triggered upon changes to the `documentation` path.

Main CI only starts the mkdocs workflow for tagged releases (using the conditionals). For the changes to latest version of docs the trigger in the workflowfile itself is used and it will run independently of the main CI (so quick patches to docs are rolled out quicker). 

Maybe I should change the job names here to `mkdocs-tag` or `mkdocs-release` to better reflect this, same for `typedocs`

> Shouldn't this have also triggered the mkdocs ? (or are we seeing an example where the `documentation/` folder has not been changed?

So in the case you mentioned above, the docs werent triggered because the commit only changed CI files. In the case when documentation gets changed, the workflow will also be separate (not viewed as part of the Main CI graph).

Reply:
>Maybe I should change the job names here to mkdocs-tag or mkdocs-release to better reflect this, same for typedocs

Fine by me, it can only help us understand the main flow better.